packages/core-web/src/Wallet.ts

```diff
@@ -0,0 +1,177 @@
+import { RpcClient } from './rpc'
+import {
+  BalanceService,
+  MintService,
+  LightningService,
+  FederationService,
+  RecoveryService,
+  WalletService,
+} from './services'
+import { logger } from './utils/logger'
+import { generateUUID } from './utils/uuid'
+
+export class Wallet {
+  public readonly id: string
+  private _client: RpcClient
+  private _clientName: string
+  private _federationId?: string
+  private _isOpen: boolean = false
+  private _openPromise: Promise<void> | undefined = undefined
+  private _resolveOpen: () => void = () => {}
```

Comment:
why does this still exist, didn't we agree that Wallet object will refer to an opened falled.